src/parser/style-context.ts

```diff
@@ -0,0 +1,94 @@
+import type { Node, Parser, Root } from "postcss";
+import postcss from "postcss";
+import { parse as SCSSparse } from "postcss-scss";
+
+import type { Context } from "../context";
+import type { SourceLocation, SvelteStyleElement } from "../ast";
+
+export interface StyleContext {
+  sourceLang: string | null;
+  sourceAst: Root | null;
+}
```

Comment:
I think it's better to change the type to the following to make the state easier to understand. What do you think?

```suggestion
export interface StyleContextSuccess {
  sourceLang: string | null;
  sourceAst: Root;
  error: false;
  unknown?: false;
}
export interface StyleContextError {
  sourceLang: string | null;
  sourceAst: null;
  error: true;
  unknown?: false;
}
export interface StyleContextNone {
  sourceLang: null;
  sourceAst: null;
  error?: false;
  unknown?: false;
}
export interface StyleContextUnknown {
  sourceLang: string;
  sourceAst: null;
  error?: false;
  unknown: true;
}
export type StyleContext = StyleContextSuccess | StyleContextError | StyleContextNone | StyleContextUnknown
```

I think the new rule should stop all reports if `StyleContextError`, or `StyleContextUnknown`.

Reply:
Great idea!
I modified it a bit to be a proper discriminated union (mostly so `if (context.status === "success")` would be a happy-path for typing), please take a look and let me know if I missed anything ;)

Reply:
It looks good to me! 👍